src/dm-area/index.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from "react";
+import { DmMap } from "./dm-map";
+
+const uploadFile = async file => {
+  const formData = new FormData();
+
+  formData.append("file", file);
+  const res = await fetch("/upload", {
+    method: "POST",
+    body: formData
+  });
+  return res.ok;
+};
+const checkMapExistence = async () => {
+  const res = await fetch("/dm/map");
+  return res.ok;
+};
+
+const Dropzone = ({ onSelectFile }) => {
+  const [isFileDragInProcess, setIsFileDragInProcess] = useState(false);
+
+  return (
+    <label
+      style={{
+        display: "block",
+        minHeight: 150,
+        border: "2px solid rgba(0,0,0,0.3)",
+        background: "white",
+        padding: "20px 20px",
+        cursor: "pointer"
+      }}
+      onDragOver={ev => {
+        setIsFileDragInProcess(true);
+        ev.stopPropagation();
+        ev.preventDefault();
+      }}
+      onDragEnd={() => {
+        setIsFileDragInProcess(false);
+      }}
+      onDragLeave={() => {
+        setIsFileDragInProcess(false);
+      }}
+      onDrop={ev => {
+        ev.preventDefault();
+        setIsFileDragInProcess(false);
+        if (ev.dataTransfer.items) {
+          const files = Array.from(ev.dataTransfer.items)
+            .filter(item => {
+              return item.kind === "file";
+            })
+            .map(item => item.getAsFile())
+            .filter(Boolean);
+
+          const [file] = files;
+          if (!file) {
+            return;
+          }
+          onSelectFile(file);
+        }
+      }}
+    >
+      <input
+        style={{ opacity: 0, width: 0, height: 0, overflow: "hidden" }}
+        accept=".jpeg,.jpg,.svg,.png"
+        type="file"
+        id="upload"
+        className="dropzone"
+        onChange={({ target }) => {
+          if (target.files) {
+            const files = Array.from(target.files);
+            const [file] = files;
+            if (!file) {
+              return;
+            }
+            onSelectFile(file);
+          }
+        }}
+      />
+      <div style={{ marginTop: "45px", textAlign: "center" }}>
+        {isFileDragInProcess
+          ? "Drop your file here"
+          : "Click here or drag and drop an image to upload"}
+      </div>
+    </label>
+  );
+};
+
+export const DmArea = () => {
+  const [hasMap, setHasMap] = useState(false);
+
+  useEffect(() => {
+    checkMapExistence().then(hasMap => {
+      if (hasMap) {
+        setHasMap(true);
+      }
+    });
+  }, []);
+
+  return (
```

Comment:
So this is the famous react!

Reply:
This, in particular, is a relatively new feature called hooks. Yuu can read more about them here: https://reactjs.org/docs/hooks-intro.html

I hope I am not shying you away from contributing if you have any questions feel free to ask 🙂

Reply:
You are good, a lot of people use react, I do not think it will be to hard to contribute. 

I look forward to asking questions and learning.